File: maria/python/update/maria_update.py
#! /usr/bin/python
# -*- coding: utf-8 -*-
#
#	maria_update.py
#
#					Jul/31/2017
#
# --------------------------------------------------------
import	sys
import	string
import	mysql.connector
#
sys.path.append('../../../common/python_common')
from sql_manipulate import sql_update_proc
#
# --------------------------------------------------------
sys.stderr.write("*** 開始 ***\n")
#
key_in = sys.argv[1]
population_in = int(sys.argv[2])
print("%s\t%d" % (key_in, population_in))
#
#
host_aa='localhost'
data_base = 'city'
user_aa ='scott'
password_aa = 'tiger123'
conn = mysql.connector.connect(user=user_aa, password=password_aa, \
                              host=host_aa,database=data_base)
#
cursor = conn.cursor()
#
sql_update_proc(cursor,key_in,population_in)
#
conn.commit()
cursor.close()
conn.close()
#
sys.stderr.write("*** 終了 ***\n")
#
# --------------------------------------------------------

File: maria/ruby/update/maria_update.rb
#! /usr/bin/ruby
# -*- coding: utf-8 -*-
#
#	maria_update.rb
#
#					Apr/07/2017
#
require 'mysql'
require 'date'
#
# ------------------------------------------------------------
def	update_proc (connection,id,population)
	date_mod=Date.today
	sql_str="UPDATE cities SET population='#{population}', DATE_MOD='#{date_mod}' where ID = '#{id}'"
	connection.query(sql_str)
end
# ------------------------------------------------------------
puts "*** 開始 ***"
#
id_in = ARGV[0]
population_in = ARGV[1].to_i
#
puts id_in,population_in
#
host = "127.0.0.1"
user = "scott"
password = "tiger123"
data_base = 'city'
connection = Mysql::new(host, user,password,data_base)
#
update_proc(connection,id_in,population_in)
#
connection.commit
connection.close
#
puts "*** 終了 ***"
# ------------------------------------------------------------
